deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/CheckDeployEnvStatusWorker.java

```diff
@@ -0,0 +1,127 @@
+package com.pinterest.teletraan.worker;
+
+import com.pinterest.deployservice.ServiceContext;
+import com.pinterest.deployservice.bean.EnvironBean;
+import com.pinterest.deployservice.dao.DeployDAO;
+import com.pinterest.deployservice.dao.EnvironDAO;
+import com.pinterest.deployservice.dao.UtilDAO;
+import com.pinterest.deployservice.bean.DeployFilterBean;
+import com.pinterest.deployservice.db.DeployQueryFilter;
+import com.pinterest.deployservice.bean.DeployQueryResultBean;
+import com.pinterest.deployservice.bean.*;
+import org.quartz.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.DataOutputStream;
+import java.net.Socket;
+import java.sql.Connection;
+import java.util.*;
+
+public class CheckDeployEnvStatusWorker implements Job {
+    private static final Logger LOG = LoggerFactory.getLogger(CheckDeployEnvStatusWorker.class);
+    private EnvironDAO environDAO;
+    private DeployDAO deployDAO;
+    private UtilDAO utilDAO;
+    public static final int  METRICS_AGENT_PORT = 18126;
+
+    public CheckDeployEnvStatusWorker() {
+
+    }
+
+    void processDeploys() throws Exception {
+        List<String> envIds = environDAO.getAllEnvIds();
+        Collections.shuffle(envIds);
+        //count for different deploy states, need emit them to statsboard
+        int runningDeploys = 0;
+        int failingDeploys = 0;
+        int succeedingDeploys = 0;
+        int succeededDeploys = 0;
+        int abortedDeploys = 0;
+
+        for (String envId : envIds) {
+            EnvironBean envBean = environDAO.getById(envId);
+            int index = 1;
+            int size = 100;
+            DeployFilterBean filterBean = new DeployFilterBean();
+            filterBean.setEnvIds(Arrays.asList(envBean.getEnv_id()));
+            filterBean.setPageIndex(index);
+            filterBean.setPageSize(size);
+            int maxPages = 50; // This makes us check at most 5000 deploys
+            int tocheckPages = maxPages;
+            while (tocheckPages-- > 0) {
+                DeployQueryFilter filter = new DeployQueryFilter(filterBean);
+                DeployQueryResultBean resultBean = deployDAO.getAllDeploys(filter);
+                if (resultBean.getTotal() < 1) {
+                    LOG.warn("Could not find any previous succeeded deploy in env {}", envBean.getEnv_id());
+                }
+                for (DeployBean deploy : resultBean.getDeploys()) {
```

Comment:
Good point, normally worker will not take more than 5 min I think.
But I do agree your concerns.
I will test in integ envrionment and calculate the time and latency for this worker since local cannot run it.